server/metadata/tenant.go

```diff
@@ -166,6 +166,39 @@ func (m *TenantManager) CreateOrGetTenant(ctx context.Context, txMgr *transactio
 	return m.createOrGetTenantInternal(ctx, tx, namespace)
 }
 
+// CreateTenant is a thread safe implementation of creating a new tenant. It returns the error if it already exists.
+func (m *TenantManager) CreateTenant(ctx context.Context, txMgr *transaction.Manager, namespace Namespace) (tenant *Tenant, err error) {
```

Comment:
do we need to return the tenant object from here?

Reply:
Could it be that the tenant is returned because it's potentially mutated during the call (namespace or other details added)?

Reply:
updated the signature to not return tenant.